### ISSUE TITLE
Various usability questions

### ISSUE THREAD

**Issue by barotto** [barotto/IBMulator#44]
Now, I have 3 questions:
1) What kind of shaders can I use? What kind of extension? i.e. .glsl is valid? .fx is valid?
2) In order to play Monkey Island, do I have to create an .img file with all the files of the game?
3) I use the 2121 BIOS but the 4-quad GUI doesn't show up like in 2011. Instead, a "Non system-disk or disk error. Press Enter" appears after memory test. What do I have to do?

_Originally posted by @Chris12121978 in https://github.com/barotto/IBMulator/issues/43#issuecomment-485077735_

**Comment by barotto** [comment]
> What kind of shaders can I use? What kind of extension? i.e. .glsl is valid? .fx is valid?

You can use one of the shaders available in share/ibmulator/gui/shaders
At the moment there are only 2 possible options:
  1. use the default shaders (the ones initially set in the ini file), which do simple image scaling
  2. use the fb-scanlines.fs fragment shader, which is the CRT styled scan-line shader by Timothy Lottes

The scaling quality is also determined by the *_filter options.

So suppose you want to use the scanline shader:
`normal_shader=gui/shaders/fb-scanlines.fs`
This will apply the scanline shader for the normal and compact GUI modes.

If you are an expert GLSL programmer you can create one youself starting from the default shaders...

> In order to play Monkey Island, do I have to create an .img file with all the files of the game?

IBMulator is not simply an emulator (like DOSBox), it's also a simulator, so it operates using disk images, for both the HDD and floppy disks.

In order to play a game:
  1. prepare a HDD image: dowload a premade one (search on my site....) or create a new one. If you decide for the latter, IBMulator will create an empty new disk image for you; you'll have to then install DOS using the DOS install floppy images or you can restore the original IBM software from a restore floppy disk set (again, look on my site).
  2. install the game you want to play from its floppy images into the HDD. (floppy images for the various games can be located on the net...) You can also copy a game's files directly on the HDD's .img using special software capable of manipulating img files: on Linux this is relatively easy, look at the "HDD image" section of README.md

Using floppy images can (and should) be done through the GUI: click on the floppy drive and a selection window will appear, pointing to the media directory where the floppy images should be located (the default media directory can be set in the ini file under the [program] section).

So at the end of the day, using IBMulator is like using a real IBM PS/1 computer. Sometimes it's incovenient, just like the real thing!

> I use the 2121 BIOS but the 4-quad GUI doesn't show up like in 2011. Instead, a "Non system-disk or disk error. Press Enter" appears after memory test. What do I have to do?

The PS/1 2011 has a so-called romdrive in the ROM image with DOS and the 4-quad software in it. The 2121 has 2 versions of the ROM, one with the romdrive and one without.

You need to use the 386/16 version of the ROM which has the romdrive built in, or, if you want to use the 386/20 version, you need a HDD image with the PS/1 software installed.

**Comment by Chris12121978** [comment]
I am not a programmer **at all**, so I guess I will just stick with the default shaders...... :)
I am 100% sure that I'm using the correct (386/16 U.S.) version, yet no GUI appears....

**Comment by barotto** [comment]
> I am 100% sure that I'm using the correct (386/16 U.S.) version, yet no GUI appears....

In this case, don't use the FC000.BIN file alone, keep the ROM zip package as is, like
`romset=PS1_2121_ROM_US.zip`

**Comment by Chris12121978** [comment]
![Χωρίς τίτλο](https://user-images.githubusercontent.com/49725551/56458582-5ccf2d80-6391-11e9-9762-a56ebb54d594.png)

**Comment by Chris12121978** [comment]
It doesn't work :(

**Comment by barotto** [comment]
Mhmm... Use dos401_2121.img floppy image, run customiz.exe and make sure
the system starts from the built in interface.

Il sab 20 apr 2019, 16:26 Chris12121978 <notifications@github.com> ha
scritto:

**Comment by Chris12121978** [comment]
Yes, it worked :) 👍 

Now....one....final question.....

Cause I am a bit greedy (no comment) I downloaded the version of ROM for the 2121 w/o the ROM drive

This one >>>>> PS1_2121_ROM.zip – P/N 93F2455, w/o ROM drive (used on US and international models)

since I wanted to use a newer BIOS and 386DX/20 Mhz

I deleted all files in the directoty where ibmulator.ini existed and started the procedure again...new....fresh....

Then, I downloaded this image.... PS1_2121_HDDFILES_WIN_US.zip – HDD preloaded software w/ Windows, US version
and used a program to insert the files contained in the zip to the **hdd-type43-PS1_2121-A82.img** file that your emulator created (you know.....the empty HDD)

And now, here is my question.

How am I supposed to boot into DOS?????? (included in the HDD)

Running customiz.exe and changing system to start from the built-in interface doesn' t work (probably cause the version of BIOS I've downloaded has no ROM drive built-in, right? Built-in means ROM drive, right?)

I promise this was the last question :)

You will get rid of me after we solve that problem......... :D

**Comment by barotto** [comment]
> Running customiz.exe and changing system to start from the built-in interface doesn' t work (the version of BIOS I've downloaded has no ROM-drive built-in and built-in = ROM-drive, right?)

Right. If you copy the files directly on the HDD you'll need to make the HDD bootable, and in order to do that you'll need to transfer the IBM PC-DOS 4 system files on the HDD with the `sys` program.

  1. boot the system with  dos401_2121.img
  2. go to the prompt and type `sys c:`

Remember, this is the real deal, you are using a PC simulator, you need to think like you used to 30 years ago :)

**Comment by Chris12121978** [comment]
Marco! Marco! 

What a blast from the past was that??????!!!!!!!   :o 

You see, back in Christmas of 1990, when my parents bought me my first PC (an IBM compatible with no instructions manual), I used to buy and read every computer magazine that was published at the time, line-by-line, word-by-word. I used to know so many DOS command, that I was believed among my friends I was computer guru. 

Nowdays, the only commands I remember and use are "dir/w" and "dir.."  :D :D 

Yes, everything worked as expected! 

Thank you.

Keep up the good work.

P.S. IMHO, the only thing you need to work a bit more (in order to have the closet experience to the real machine), is CRT-shaders.

**Comment by barotto** [comment]
> IMHO, the only thing you need to work a bit more (in order to have the closet experience to the real machine), is CRT-shaders.

This is on the TODO list. So many ideas, so little free time :(

> And needs a better optimization in the code cause Mr. Lottes scanlines shader (at least on my old computer), is very slaggy. It makes emulator almost unusable.

What are your PC specs? I've tested on anything other than my PC with a GTX 970, which by today standards is not very fast, but still an order of magnitude faster than any integrated GPU.

**Comment by Chris12121978** [comment]
![Χωρίς τίτλο](https://user-images.githubusercontent.com/49725551/56508632-71482d00-652d-11e9-9b10-b92a7c4cfe70.png)

**Comment by Chris12121978** [comment]
I don't think it's about computer specs. On DosBox SVN (Daum version) 99% of shaders work pretty fast on the same machine. (i.e. CRT-geom-curved.fx which is a very good replica of a 90's CRT monitor, runs very fast.)

**Comment by Chris12121978** [comment]
How old are you?
Are you Italian?
Do you work as a full-time programmer?
